src/main.cpp

```diff
@@ -108,106 +69,240 @@ class DMG {
 
 		return true;
 	}
+
+	bool tick_frame() {
+		do {
+			tick();
+		} while (!bus.ppu.frame_complete);
```

Comment:
Could be issues with running a frame at a time once I need to handle screen interrupts for mode 7 style graphics.